bundles/binding/org.openhab.binding.mystromecopower/src/main/java/org/openhab/binding/mystromecopower/internal/api/MystromClient.java

```diff
@@ -0,0 +1,304 @@
+/**
+ * Copyright (c) 2010-2014, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.mystromecopower.internal.api;
+
+import org.openhab.binding.mystromecopower.internal.api.model.*;
+import org.slf4j.Logger;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import java.util.ArrayList;
+import java.util.List;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+/**
+ * Manage Json Api call to mystrom Api.
+ * @since 1.7.0-SNAPSHOT
+ * @author Jordens Christophe
+ *
+ */
+public class MystromClient implements IMystromClient{
+	private Logger logger;
+	private JsonParser jsonParser;
+	private Gson gson;
+	private String userName;
+	private String password;
+	private String authToken;
+	private static final String API_URL = "https://mystrom.ch/mobile/";
+	
+	/**
+	 * Initialize the MystromClient class.
+	 * @param userName The user name for the mystrom server connection.
+	 * @param password The password for the mystrom server connection.
+	 * @param logger The logger used to log into openhab.
+	 */
+	public MystromClient(String userName, String password, Logger logger)
+	{
+		this.gson = this.createGsonBuilder().create();
+		this.logger = logger;
+		this.userName = userName;
+		this.password = password;
+		this.jsonParser = new JsonParser();
+	}
+	
+	/* (non-Javadoc)
+	 * @see org.openhab.binding.mystromecopower.internal.api.IMystromClient#login()
+	 */
+	public Boolean login(){
+		Reader reader = null;
+		this.logger.debug("Do login for user '{}'", this.userName);
+		
+		try {
+			HttpURLConnection httpURLConnection;
+			httpURLConnection = (HttpURLConnection) new URL(API_URL + "auth?email="
```

Comment:
Not correct for HTTPS connections.  The URI lives in the header which is encrypted just as the body.